ckanext/s3filestore/uploader.py

```diff
@@ -151,12 +155,29 @@ def get_signed_url_to_key(self, key_path, expiredin=3600):
         # check whether the object exists in S3
         client.head_object(Bucket=self.bucket_name, Key=key_path)
 
+        current_time = time.time()
+        redis_conn = connect_to_redis()
+        cache_key = REDIS_PREFIX + key_path
+        cache_value = redis_conn.get(cache_key)
+        if cache_value:
+            cache_time, cache_url = cache_value.split('|', 1)
+            cache_time = int(float(cache_time))
+            if current_time - cache_time < self.signed_url_cache_window:
```

Comment:
should it not expire in cache and not come up?

Reply:
Actually I haven't been using Redis' built-in expiry, but that would probably be a better approach.

Reply:
can set expire on a key
 def expire(self, name, time):
        """
        Set an expire flag on key ``name`` for ``time`` seconds. ``time``
        can be represented by an integer or a Python timedelta object.
        """
        if isinstance(time, datetime.timedelta):
            time = int(time.total_seconds())
        return self.execute_command('EXPIRE', name, time)

    def expireat(self, name, when):
        """
        Set an expire flag on key ``name``. ``when`` can be represented
        as an integer indicating unix time or a Python datetime object.
        """
        if isinstance(when, datetime.datetime):
            when = int(mod_time.mktime(when.timetuple()))
        return self.execute_command('EXPIREAT', name, when)

Reply:
https://github.com/andymccurdy/redis-py/blob/master/redis/client.py#L1645

Reply:
or on the set itself.

 def set(self, name, value,
            ex=None, px=None, nx=False, xx=False, keepttl=False):
        """
        Set the value at key ``name`` to ``value``
        ``ex`` sets an expire flag on key ``name`` for ``ex`` seconds.
        ``px`` sets an expire flag on key ``name`` for ``px`` milliseconds.
        ``nx`` if set to True, set the value at key ``name`` to ``value`` only
            if it does not exist.
        ``xx`` if set to True, set the value at key ``name`` to ``value`` only
            if it already exists.
        ``keepttl`` if True, retain the time to live associated with the key.
            (Available since Redis 6.0)
        """

Reply:
> Actually I haven't been using Redis' built-in expiry, but that would probably be a better approach. 

yep, then it self cleanses.